app/templates/components/forms/discount-codes/create-discount-codes-form.hbs

```diff
@@ -0,0 +1,102 @@
+    {{ui-radio label=(t 'Amount (US$)') name='discount_type'  value='amount' current='amount' onChange=(action (mut selectedMode))}}
+  </div>
+  {{#if (not-eq selectedMode 'percent')}}
+    <div class="field">
+      {{input type='number' name='discount_amount' min=1}}
+    </div>
+  {{/if}}
+  <div class="field">
+    {{ui-radio label=(t 'Percent (%)') name='discount_type'  value='percent' onChange=(action (mut selectedMode))}}
+  </div>
+  {{#if (eq selectedMode 'percent')}}
+    <div class="field">
+      {{input type='number' name='discount_percent' min=1 max=100}}
+    </div>
+  {{/if}}
+
+  <div class="field">
+    <label class="required">{{t 'Number of Discount Tickets'}}</label>
+    {{input name='number_of_discount_tickets' type='number' value='10' min='0'}}
+  </div>
+
+  <div class="grouped inline fields">
+    <label class="required">{{t 'Status'}}</label>
+    <div class="field">
+      {{ui-radio name='status' label='Active' current='active' value='active'}}
+    </div>
+    <div class="field">
+      {{ui-radio name='status' label='Inactive' value='inactive'}}
+    </div>
+  </div>
+
+  <div class="field">
+    <label class="required">{{t 'Select Ticket(s) applied to the discount code'}}</label>
+    <div class="ui checkbox">
+      {{ui-checkbox label="Select all Ticket types" checked=checked onChange=(action (mut checked))}}
+    </div>
+  </div>
+
+  <div>
+    <div class="inline field">
+      <label>{{t 'Show more options'}}</label>
+      {{ui-checkbox class='toggle' checked=showMoreOptions onChange=(action (mut showMoreOptions))}}
+    </div>
+    {{#if showMoreOptions}}
+
+      <div>{{t 'Set the min and max quantity allowed to purchase per order (Optional)'}}</div>
+      <div class="fields">
+        <div class="wide field {{if device.isMobile 'sixteen' 'five'}}">
+          <div class="ui labeled input">
+            <div class="ui label">
+              {{t 'Min'}}
+            </div>
+            {{input type='number' min='0'}}
+          </div>
+        </div>
+        <div class="wide field {{if device.isMobile 'sixteen' 'five'}}">
+          <div class="ui labeled input">
+            <div class="ui label">
+              {{t 'Max'}}
+            </div>
+            {{input type='number' min='0'}}
+          </div>
+        </div>
+      </div>
+
+
+      {{t 'Validity Period(Optional)'}}
+      <div class="fields">
+        <div class="wide field {{if device.isMobile 'sixteen' 'five'}}">
+          <label>{{t 'Valid from'}}</label>
+          {{widgets/forms/date-picker}}
+        </div>
+        <div class="wide field {{if device.isMobile 'sixteen' 'five'}}">
+          <label>{{t 'Expires on'}}</label>
+          {{widgets/forms/date-picker}}
+        </div>
+        <div class="ui hidden divider"></div>
+      </div>
+    {{/if}}
+    <div class="field">
+      <label>{{t 'Discount Link'}}</label>
+      {{input type='url' name='discount_link' class='field disabled'}}
```

Comment:
This should be auto generated when user enters the discount code, please check the behaviour on the orga-server